kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/client/DefaultTableClient.java

```diff
@@ -26,7 +26,7 @@ public class DefaultTableClient
     implements TableClient {
     private final Configuration hadoopConf;
 
-    private DefaultTableClient(Configuration hadoopConf) {
```

Comment:
Making this public is necessary if we want to make it easy for connectors to selectively override the handlers.

Reply:
I can do this in a separate PR however

Reply:
If we make this public, the static create methods looks odd. This class is just a few one-liner methods, copying shouldn't be an issue. Can we hide this from public API docs? or make this `protected`?

Reply:
`protected` works and I checked the java docs it's excluded